src/util/ConfigManager/ConfigManager.h

```diff
@@ -40,6 +41,60 @@ namespace ad_utility {
 Manages a bunch of `ConfigOption`s.
 */
 class ConfigManager {
+  /*
+  Our hash map always saves either a configuration option, or a sub manager
+  (another `ConfigManager`).
+
+  This class makes the handeling more explicit and allows for more information
+  to be saved alongside the configuration options and sub managers.
+  For example: The order, in which they were created.
+  */
+  class HashMapEntry {
+   public:
+    using Data = std::variant<ConfigOption, ConfigManager>;
+
+   private:
+    std::unique_ptr<Data> data_;
```

Comment:
Why is this a `unique_ptr`?

Reply:
`HashMapEntry` exists for the better management of entries in our hash map and the hash map has ownership over created configuration options, and sub manager. Multiple functions give back references to those configuration options and sub managers.

But, because adding new configuration options and sub managers to the hash map could destroy the integrity of those references, the hash map only manages the unique pointer, that actually have ownership.

This of course means, that either the hash map must contain `std::unique_ptr<HashMapEntry` (And the `HashMapEntry` uses a normal variant for `data_`), or the `HashMapEntry` contains/manages the `std::unique_ptr`.

I thought, that inside the `HashMapEntry` would be a better place, because then you can just use move operators with the hash map entries. Which is just way easier.